### PR TITLE
Hn style homepage

### PR DESCRIPTION
**Pull request by hungnguyen1895**

### CODE REVIEW

File: src/components/Beat.tsx [hunk 1]
```diff
@@ -4,6 +4,13 @@ import { useState, useEffect } from 'react';
 import { BackendContext } from '../util/api';
 import {PlaylistDetail, Song} from '../util/api/types';
```
Comment (lloydoad) [inline src/components/Beat.tsx]: Nit: inconsistent spacing

File: src/components/Beat.tsx [hunk 2]
```diff
@@ -24,12 +31,13 @@ const useStyles = makeStyles(() => ({
...
     position: 'relative',
+    cursor: 'pointer'
   },
   background: {
     backgroundRepeat: 'no-repeat',
-    width: 300,
-    height: 250,
-    opacity: 0.4
+    width: 200,
+    height: 150,
```
Comment (lloydoad) [inline src/components/Beat.tsx]: Nit: specify px/pc/etc

File: src/components/Beat.tsx [hunk 3]
```diff
@@ -56,18 +65,22 @@ const useStyles = makeStyles(() => ({
     padding: 2,
     backgroundColor: 'grey',
     margin: 5,
-    opacity: 0.7
+    opacity: 0.7,
+    fontSize: 10
```
Comment (lloydoad) [inline src/components/Beat.tsx]: Nit: specify unit

File: src/components/Beat.tsx [hunk 4]
```diff
@@ -56,18 +65,22 @@ const useStyles = makeStyles(() => ({
...
   },
 }));
 
-const Beat: React.FC = () => {
+const Beat: React.FC<BeatProps> = ({...props}) => {
   const api = React.useContext(BackendContext);
+  const musicProvider = React.useContext(MusicContext);
+
```
Comment (lloydoad) [inline src/components/Beat.tsx]: Spacing

File: src/components/LoginForm.tsx [hunk 5]
```diff
@@ -61,10 +56,24 @@ const LoginForm: React.FC<LoginProps> = ({ ...props }) => {
...
+      try {
+        history.replace('/');
+      } catch (e) {
+        alert(e.message);
+      }
+  };
+
 	return (
 	  <>
     <CssBaseline />
```
Comment (lloydoad) [inline src/components/LoginForm.tsx]: Inconsistent tabs and spacing. I don't understand what's happening here

File: src/components/LoginForm.tsx [hunk 6]
```diff
@@ -129,11 +154,24 @@ const LoginForm: React.FC<LoginProps> = ({ ...props }) => {
...
+          <button onClick={() => (
+
+            // save user to DB
+
+            loginWithRedirect()
+
+          )}>
+            Log in with Auth0
+          </button>         
+        </Container> */}
```
Comment (lloydoad) [inline src/components/LoginForm.tsx]: Not sure if you intended to upload this or not

File: src/components/NavBar.tsx [hunk 7]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { makeStyles } from '@material-ui/core/styles';
 import clsx from 'clsx';
+import { Link } from 'react-router-dom';
 
 const useStyles = makeStyles(() => ({
   navbar: {
-    height: 40,
+    height: 53,
```
Comment (lloydoad) [inline src/components/NavBar.tsx]: @see previous comment

File: src/components/NavBar.tsx [hunk 8]
```diff
@@ -19,6 +19,10 @@ const useStyles = makeStyles(() => ({
   active: {
     backgroundColor: '#003585'
   },
+  tab: {
+    color: 'white',
```
Comment (lloydoad) [inline src/components/NavBar.tsx]: `white` constant can change. Use `#fff` if possible instead

File: src/components/NavBar.tsx [hunk 9]
```diff
@@ -28,8 +32,12 @@ const NavBar: React.FC = () => {
   return(
     
     <ul className={classes.navbar}>
-      <li className={clsx(classes.navbarElement, classes.active)}><a href=''>Home</a></li>
-      <li className={clsx(classes.navbarElement)}><a href=''>Studio</a></li>
+      <li className={clsx(classes.navbarElement, classes.active)}>
+        <Link to="/" className={classes.tab}>
+          Home 
+          <img style={{width: 20, height: 20, paddingLeft: 5}} src='images/HomeImage.png' alt=''></img>
```
Comment (lloydoad) [inline src/components/NavBar.tsx]: @see previous comment about units

Comment (lloydoad) [reply src/components/NavBar.tsx]: Also better to store image locations are constants at top of the file. It makes it easier to change/update in the future without searching through the entire file

File: src/components/Playlists.tsx [hunk 10]
```diff
@@ -10,6 +10,7 @@ import {
     CardMedia,
     Container
 } from '@material-ui/core/';
+import MusicContext from '../util/contexts/music/MusicContext';
```
Comment (lloydoad) [inline src/components/Playlists.tsx]: Maybe move to line 4? Idk, just seems better stylewise

File: src/components/PrivateRoute.tsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+} from "react-router-dom";
+
+interface PrivateRouteProps extends RouteProps {
+  isAuthenticated: boolean;
+// Would reccommend this comes from a global state manager
+// such as redux, shown using basic props here
+}
+
+export class PrivateRoute extends Route<PrivateRouteProps> {
+
```
Comment (lloydoad) [inline src/components/PrivateRoute.tsx]: Nit: unnecessary new line

File: src/components/PrivateRoute.tsx [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+                  return React.createElement(this.props.component);
+              } 
+
+              if(this.props.render) {
+                  return this.props.render(props);
+              }
+          }} />
+      );
+  }
+}
```
Comment (lloydoad) [inline src/components/PrivateRoute.tsx:34]: No newline at end of file

File: src/components/PublicBeat.tsx [hunk 13]
```diff
@@ -17,36 +24,51 @@ const useStyles = makeStyles(() => ({
...
-    width: 300,
-    height: 250,
+    width: 200,
+    height: 150,
     opacity: 0.4
   },
   bottomLeftCorner: {
     position: 'absolute',
-    bottom: 0
+    bottom: 0,
```
Comment (lloydoad) [inline src/components/PublicBeat.tsx]: @see comments previous about units

File: src/components/PublicBeat.tsx [hunk 14]
```diff
@@ -17,36 +24,51 @@ const useStyles = makeStyles(() => ({
...
 
   let beats = [
     {
+      'id': '6',
       'picture': 'https://upload.wikimedia.org/wikipedia/commons/thumb/3/37/African_Bush_Elephant.jpg/440px-African_Bush_Elephant.jpg',
       'background': 'https://img-aws.ehowcdn.com/560x560p/s3-us-west-1.amazonaws.com/contentlab.studiod/getty/aac4f9b5127946ec8cc85c718d4261d5',
       'title': 'guitar testing title',
-      'type': 'abc',
+      'type': 'abcabcabcabcabcabcabcabcabc',
       'numBeats': 10,
```
Comment (lloydoad) [inline src/components/PublicBeat.tsx]: This seems like test data. Usually better to store this in an external JSON file to make it easier to understand the actual code

File: src/components/PublicBeat.tsx [hunk 15]
```diff
@@ -99,21 +132,20 @@ const PublicBeat: React.FC = () => {
       <div className={classes.scroll}>
         {beats.map((beat, key) => {
           return (
-            <div className={classes.card} key={key}>
+            <div className={classes.card} key={key} onClick={() => playPublicBeat(beat.id)}>
               <img alt='' className={classes.background} src={beat.background}></img>
               <div className={classes.bottomLeftCorner}>
-                <img alt='' style={{float: 'left'}} className={classes.beatPicture} src={beat.picture}></img>
-                <div style={{float: 'left'}}>
+                <img className={classes.beatPicture} src={beat.picture} alt=""></img>
```
Comment (lloydoad) [inline src/components/PublicBeat.tsx:82]: Accessibility: No alt given

File: src/components/PublicBeat.tsx [hunk 16]
```diff
@@ -99,21 +132,20 @@ const PublicBeat: React.FC = () => {
...
                       <div>{beat.type}</div>
                       <div>{beat.duration}</div>
                     </div>
                   </div>
                   </div>
-                </div >
+                </div>
             </div>
           )
         })}
```
Comment (lloydoad) [inline src/components/PublicBeat.tsx]: I'm not sure but it looks like 4-spaced tabs and 2-spaced tabs are being mixed here

File: src/components/PublicBeatPlayButton.tsx [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  return (
+    <>
+    {button}
+    </>
+  );
+}
+
+export default PublicBeatPlayButton;
```
Comment (lloydoad) [inline src/components/PublicBeatPlayButton.tsx]: No newline at end of file

File: src/components/PublicBeatPlayButton.tsx [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+  } else {
+    button = (
+      <img className={classes.playButton} alt='' src='images/playButton.png' onClick={playOrPause}></img>
+    );
+  }
+
+  return (
+    <>
+    {button}
+    </>
```
Comment (lloydoad) [inline src/components/PublicBeatPlayButton.tsx]: What's `<>`?

File: src/components/SideBar.tsx [hunk 19]
```diff
@@ -1,78 +1,262 @@
...
 const useStyles = makeStyles(() => ({
-  sidebar: {
-    height: '100%',
-    color: 'white',
+  fixedLeftBottom: {
+    position: 'fixed',
+    bottom: 0,
+    width: '100%',
     backgroundColor: '#1a1919',
   },
```
Comment (lloydoad) [inline src/components/SideBar.tsx]: Can these `useStyles` object be put in their own files? They seem big enough to warrant that

File: src/components/SideBar.tsx [hunk 20]
```diff
@@ -1,78 +1,262 @@
...
+          'audioUrl': 'http://www.hubharp.com/web_sound/WalloonLilliShort.mp3',
+          'title': 'audio 1 title',
+          'authorName': 'author 1'
+        },
+        {
+          'imageUrl': 'https://qph.fs.quoracdn.net/main-qimg-70d48d9e6d598aa364c13ef739b489d4',
+          'audioUrl': 'http://www.hubharp.com/web_sound/BachGavotteShort.mp3',
+          'title': 'audio 2 title',
+          'authorName': 'author 2'
+        }
```
Comment (lloydoad) [inline src/components/SideBar.tsx]: Test data should be in own json file. Makes it easier to change values for testing, especially as this grows over time

**Review by lloydoad — CHANGES_REQUESTED** [review]
Left some comments regarding style and the general architecture of how things were setup. I was only able to get to the first 2/3 of the files changed. I suspect some of the comments exist in places I did not notice so you may want to look through everything in case I missed a comment somewhere. Also there's a yarn.lock file uploaded; guessing that was not by intention.

File: src/components/PublicBeatPlayButton.tsx (hunk 18 above)
Comment (hungnguyen1895) [reply src/components/PublicBeatPlayButton.tsx]: It is called Fragment, sometimes you don't want to have so many nested components like <div>, it's hard to style or debug or going through the whole nested tree to find our components, so React has this Fragment <></> to be able to render a component without a parent tag.

**Comment by WarmBuns** [comment]
Are we applying any style configs such as prettier or anything? Might help fix the inconsistencies

**Comment by WarmBuns** [comment]
Why are there 44 files changed?

**Comment by hungnguyen1895** [comment]
> Why are there 44 files changed?

because I added a lot of pictures from Lloyd, and the home view has many components ( many files).

**Pull request merged**